### PR TITLE
Fixes #224 | Started basis of photoresistorSensor documentation

### LINKED ISSUE #224: Photoresistor Documentation Support
Documentation requirements:

- [x] link to the documentation
- [x] Overview 
- [x] Components
- [x] Assembly Instructions
- [x] Circuit Diagram
- [x] Functionality
- [x] Testing
- [x] Troubleshooting
- [x] YAML
- [x] Constructor and Methods
- [x] Example Controller

### PR DESCRIPTION
**Pull request by adelgadoj3**
Began process for PhotoResistorSensor documentation.

### CODE REVIEW

File: pi4micronaut-utils/src/docs/asciidoc/components/inputComponents/photoresistorSensor.adoc [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+
+
+==== PhotoResistor Sensor
+[.text-right]
+https://github.com/oss-slu/Pi4Micronaut/edit/develop/pi4micronaut-utils/src/docs/asciidoc/components/inputComponents/photoresistorSensor.adoc[Improve this doc]
+
+
+===== Overview
+This section provides the details of the PhotoResistor Sensor, also known as the light sensor, circuit, including its components, assembly instructions, and functionality.
```
Comment (SwindleA) [inline pi4micronaut-utils/src/docs/asciidoc/components/inputComponents/photoresistorSensor.adoc:19]: reword this sentence. One suggestion is to move the comment about the light sensor to a separate sentence.

File: pi4micronaut-utils/src/docs/asciidoc/components/inputComponents/photoresistorSensor.adoc [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+===== Components
+* 1 x RaspberryPi
+* 1 x Breadboard
+* 1 x LED
+* 1 x PhotoResistor Sensor
+* 1 x 220Ω Resistor
+* 1 x Capacitor (10µf)
+* 2 x 10KΩ Resistor
+* 4 Jumper Wires
+* Power source (appropriate voltage, typically 3.3V or 5V)
```
Comment (SwindleA) [inline pi4micronaut-utils/src/docs/asciidoc/components/inputComponents/photoresistorSensor.adoc:30]: the voltage should be specified.

File: pi4micronaut-utils/src/docs/asciidoc/components/inputComponents/photoresistorSensor.adoc [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+* Place the PhotoResistor Sensor onto the breadboard inline with the Capacitor, 10KΩ Resistor. The PhotoResistor should have a 10KΩ pin on either side of its' pins.
+* The 10KΩ resistor pin should then be placed in line away from the PhotoResistor pins.
+* Place a jumper wire from GND pin to the negative side of the board.
+* Place a jumper wire from GPIO4 next to the positive pin of the Capacitor and the 10KΩ Resistor.
+* Place a jumper wire from GPIO17 and place it next to the 220Ω Resistor
+* Place a jumper wire from GPIO27 to the pin of the 10KΩ.
+
+===== Circuit Diagram
+
+image::photoresistor-CD.png[]
```
Comment (SwindleA) [inline pi4micronaut-utils/src/docs/asciidoc/components/inputComponents/photoresistorSensor.adoc:48]: are you sure the circuit diagram is correct?  I am not sure it matches your description and you do not have  a wire connected to GPIO4. Also, the resistor on the south side of the board is connected to SPIMOSI pin and is not connect to the rest of the setup

File: pi4micronaut-utils/src/docs/asciidoc/components/inputComponents/photoresistorSensor.adoc [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+[source, bash]
+----
+$curl http://localhost:8080/photoResistor/threshold/i
+----
+
+* `/threshold/{i}` - Sets the threshold for which the LED will turn on.
+
+===== Troubleshooting
+LED not lighting:
+    * Verify that the appropriate resistors are being used (220Ω)
```
Comment (SwindleA) [inline pi4micronaut-utils/src/docs/asciidoc/components/inputComponents/photoresistorSensor.adoc:87]: for some reason, the bullet points in the troubleshooting section are not rendering properly. I think you need to remove the tab before them. 
image

**Review by SwindleA — CHANGES_REQUESTED** [review]
see the comments. Also the component needs to be added to the index.adoc so that it shows up in the documentation.

File: pi4micronaut-utils/src/docs/asciidoc/components/inputComponents/photoresistorSensor.adoc [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+[source, yaml]
+----
+  digital-input:
+    photo-resistor-input:                 # <.>
+      name: Photo Resistor Input          # <.>
+      address: 4                          # <.>
+      debounce: 100000                    # <.>
+      pull: PULL_DOWN                     # <.>
+      provider: pigpio-digital-input      # <.>
+
```
Comment (ruthvikm) [inline pi4micronaut-utils/src/docs/asciidoc/components/inputComponents/photoresistorSensor.adoc:107]: Add digital-output part of photo resistor.

File: pi4micronaut-utils/src/docs/asciidoc/components/inputComponents/photoresistorSensor.adoc [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+----
+
+
+===== Constructor and Methods
+To see the constructor and methods of the PhotoResistorHelper class see our java doc link:https://oss-slu.github.io/Pi4Micronaut/javadoc/com/opensourcewithslu/inputdevices/PhotoResistorHelper.html[here] for more details.
+
+
+===== Example
```
Comment (ruthvikm) [inline pi4micronaut-utils/src/docs/asciidoc/components/inputComponents/photoresistorSensor.adoc:115]: Example Controller

**Review by ruthvikm — CHANGES_REQUESTED** [review]

File: pi4micronaut-utils/src/docs/asciidoc/index.adoc [hunk 7]
```diff
@@ -68,6 +68,8 @@ include::components/inputComponents/touchSwitch.adoc[]
 
 include::components/inputComponents/pirSensor.adoc[]
 
+include::components/inputComponents/photoresistorSensor.adoc[]
```
Comment (SwindleA) [inline pi4micronaut-utils/src/docs/asciidoc/index.adoc:71]: remove the photosensor link in line 63, and delete the adoc file.

**Review by SwindleA — CHANGES_REQUESTED** [review]

**Review by ruthvikm — CHANGES_REQUESTED** [review]
Circuit diagram need changes.

**Review by SwindleA — APPROVED** [review]
looks fabulous

**Pull request merged**